### PR TITLE
Add chorus support to the mixer

### PR DESCRIPTION
**Pull request by johnnovak**
This implements https://github.com/dosbox-staging/dosbox-staging/issues/1680

It's the easiest to hear the effect in single-OPL games with the reverb turned off (e.g used Alone in the Dark 1, floppy version).

Some testing recommendations & demonstrations of the various settings in combination with reverb and crossfeed. All games were configured for OPL music. These are my personal configs for these games (I'm an OPL fan, as you can tell); I'm really pleased with how the new audio enhancement features work together to elevate the sound of these games to a whole new level! 

### Alone in the Dark (Floppy version)

```ini
[sblaster]
sbtype = sbpro1

[mixer]
reverb = large
chorus = normal
crossfeed = on

[autoexec]
mixer sb 85 /noshow
```

### Drakkhen

```ini
[sblaster]
sbtype = sbpro1

[mixer]
reverb = small
chorus = strong
crossfeed = on
```

### Elvira - Mistress of the Dark

```ini
[sblaster]
sbtype = sbpro1

[mixer]
reverb = medium
chorus = normal
crossfeed = on
```

### KGB

```ini
[sblaster]
sbtype = sbpro1
oplmode = opl3gold

[mixer]
reverb = small
chorus = normal

[autoexec]
mixer master 150 /noshow
```

### Quest for Glory II - Trial by Fire

```ini
[sblaster]
sbtype = sbpro1

[mixer]
reverb = large
chorus = strong
crossfeed = on
```

### Legend of Kyrandia Book 1

```ini
[sblaster]
sbtype = sbpro2

[mixer]
reverb = large
chorus = normal
crossfeed = on

[autoexec]
mixer fm 150 sb 70 /noshow
```

### Indiana Jones and the Last Crusade (EGA)
```ini
[sblaster]
sbtype = sbpro1

[mixer]
reverb = medium
chorus = strong
```

### X-COM - UFO Defense

```ini
[sblaster]
sbtype = sb16

[mixer]
reverb = huge
chorus = normal
crossfeed = on

[autoexec]
mixer master 200 /noshow
```

### Space Quest III - The Pirates of Pestulon

```ini
[sblaster]
sbtype = sbpro1
opl_filter = off

[mixer]
reverb = small
chorus = subtle
```

### Ultima Underworld - The Stygian Abyss

```ini
[sblaster]
sbtype = sbpro1
sbmixer = false
opl_filter = off

[mixer]
reverb = large
chorus = subtle
crossfeed = on

[autoexec]
mixer master 160 sb 40 /noshow
```

### CODE REVIEW

File: src/hardware/mixer.cpp [hunk 1]
```diff
@@ -260,11 +304,17 @@ static void configure_reverb(std::string reverb_pref)
...
 		reverb_pref = "medium";
 
-	enum class RoomSize { tiny, small, medium, large, huge };
-	const auto room_size = enum_cast<RoomSize>(reverb_pref).value();
-
 	auto &r = mixer.reverb; // short-hand reference
 
+	auto current_preset = r.preset;
+	auto new_preset = enum_cast<ReverbPreset>(reverb_pref).value();
+	auto preset_changed = (current_preset == new_preset);
```
Comment (kcgen) [inline src/hardware/mixer.cpp:311]: Probably meant `preset_unchanged`

Comment (johnnovak) [reply src/hardware/mixer.cpp:311]: Yeah, I remember I flipped the condition at some point :)

**Comment by johnnovak** [comment]
Ah, just remembered that I need to remove the FluidSynth reverb as well, like you did with the reverb. I'll get that done too soon.

**Comment by kcgen** [comment]
Looks good, another one down!

**Pull request merged**